ssms/views.py

```diff
@@ -1480,492 +1428,114 @@ def allocate(grub, grubstu, no, batchlist):
         if (count > batchgroup):
             j = j + 1
             count = 0
+# helper function takes arguments as gmid, request and boolean meal_type. meal_type = true for VegAllocation
+# meal_type = false for NonVegAllocation
+def batch_allocate(gmid, request, meal_type):
+    grub = Grub.objects.get(gm_id=gmid)
+    batch_list = [] #generates a batch list to be passed onto allocate(grub, grubstu, no, batchlist)
+    name_list = {1: "A", 2: "B", 3: "C", 4: "D"}
+    if meal_type:
+        #change html page inorder to map VegAllocate1,2,3 in a dict and then access it
+        if "VegAllocate1" in request.POST:
+            n = 1
+        elif "VegAllocate2" in request.POST:
+            n = 2
+        elif "VegAllocate3" in request.POST:
+            n = 3
+        else :
+            n = 4
 
+    else:
+        if "NonVegAllocate1" in request.POST:
+            n = 1
+        elif "NonVegAllocate2" in request.POST:
+            n = 2
+        elif "NonVegAllocate3" in request.POST:
+            n = 3
+        else :
+            n = 4
+
+    for i in range(1, n + 1):
+        batch = Batch.objects.create(
+            gm_id=grub,
+            meal="Veg" if meal_type else "Non Veg",
+            batch_name=name_list[i],
+            color=request.POST["batch{}color".format(i)],
+            timing=request.POST["batch{}time".format(i)],
+        )
+        batch_list.append(batch)
 
+    grubstu = Grub_Student.objects.filter(
+        gm_id=grub.gm_id, status="Signed Up").order_by('bhawan')
+    allocate(grub, grubstu, n, batch_list) #allocating the batches using the batchlist generated
+
+
+@user_passes_test(lambda user: user.is_superuser, login_url='ssms/coord/login')
 def ssms_grub_batchallocation(request, gmid):
-    if request.user.is_superuser:
```

Comment:
You removed this line. Anybody can access this page.
Please remember the access privileges while changing/creating a view.

Apart from that, nice work! Appreciate it. 😄

Reply:
fixed access in send2 and ssms_student_table in https://github.com/SSMS-Pilani/SSMS-Grub-Portal/commit/0e748ee8bcdaf5126f8ad214dc382d459bf5d8af